packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -268,20 +260,46 @@ export function markAllThreadsInTeamRead(userId: string, teamId: string) {
     };
 }
 
-export function markThreadAsUnread(userId: string, teamId: string, threadId: string, postId: string) {
+export function markThreadAsUnread(userId: string, teamId: string, threadId: string) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        try {
-            await Client4.markThreadAsUnreadForUser(userId, teamId, threadId, postId);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-            dispatch(logError(error));
-            return {error};
+        const getPostsForThread = makeGetPostsForThread();
+        let posts = getPostsForThread(getState(), threadId);
+
+        // load posts in thread if they are not loaded already
+        if (posts.length < 2) {
+            getPostThread(threadId)(dispatch, getState).then(({data, error}) => {
```

Comment:
can we change this to the most idiomatic 


```suggestion
            dispacth(getPostThread(threadId)).then(({data, error}) => {
```

Reply:
Good point, thanks